### PR TITLE
Improved formatting for index.yaml file

### PR DESCRIPTION
**Pull request by Sammers21**
Formatting and generation have been improved to look more like a file generated by `helm` utility:

1. Formatted in block style
2. Set for `urls` field has been replaced with list
3. An entry is generated if it did not exist before

Related to #8

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#46](https://github.com/artipie/helm-adapter/issues/46) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/703255c6-ca92-448a-a9be-d38d35a81056); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/helm/IndexYaml.java [hunk 1]
```diff
@@ -148,14 +155,20 @@ private void update(final Map<String, Object> index, final TgzArchive tgz) {
         final ChartYaml chart = tgz.chartYaml();
         final String version = "version";
         final Map<String, Object> entries = (Map<String, Object>) index.get("entries");
+        final String name = (String) chart.field("name");
```
Comment (Vatavuk) [inline src/main/java/com/artipie/helm/IndexYaml.java:158]: @Sammers21 extracting of field can produce a null value. Maybe we can create `stringField` method in `ChartYaml` which will throw exception if field is missing or return fields string value. wdyt?

File: src/main/java/com/artipie/helm/IndexYaml.java [hunk 2]
```diff
@@ -148,14 +155,20 @@ private void update(final Map<String, Object> index, final TgzArchive tgz) {
...
+        if (!entries.containsKey(name)) {
+            entries.put(name, new ArrayList<Map<String, Object>>(0));
```
Comment (Vatavuk) [inline src/main/java/com/artipie/helm/IndexYaml.java:160]: @Sammers21 why modifying the map?

Comment (Sammers21) [reply src/main/java/com/artipie/helm/IndexYaml.java:160]: It is the main point of the update method. We fill it with new values. If there is no such entry(the first time upload), we add a new one

File: src/main/java/com/artipie/helm/IndexYaml.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/com/artipie/helm/IndexYaml.java:158]: The package could not be produced by helm if there is no such field.

File: src/main/java/com/artipie/helm/IndexYaml.java (hunk 2 above)
Comment (Vatavuk) [reply src/main/java/com/artipie/helm/IndexYaml.java:160]: @Sammers21 ok

File: src/main/java/com/artipie/helm/IndexYaml.java (hunk 1 above)
Comment (Vatavuk) [reply src/main/java/com/artipie/helm/IndexYaml.java:158]: @Sammers21 ok

**Comment by 0crat** [comment]
Job was finished in 10 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by Vatavuk** [comment]
@g4s8 please remove this issue from scope

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**